api/provider/aws/service_list.go

```diff
@@ -12,102 +12,79 @@ import (
 // A Service Summary consists of the Service ID, Service name, Environment ID, and
 // Environment name.
 func (s *ServiceProvider) List() ([]models.ServiceSummary, error) {
-	clusterArns, err := s.listClusters()
+	clusterNames, err := listClusterNames(s.AWS.ECS, s.Config.Instance())
 	if err != nil {
 		return nil, err
 	}
 
-	clusterServices := map[*string][]*string{}
+	serviceARNs, err := s.listClusterServiceARNs(clusterNames)
+	if err != nil {
+		return nil, err
+	}
 
-	for _, clusterArn := range clusterArns {
-		clusterName := strings.Split(aws.StringValue(clusterArn), ":")[5]
-		clusterName = strings.Replace(clusterName, "cluster/", "", -1)
+	serviceIDs := make([]string, len(serviceARNs))
+	for i, serviceARN := range serviceARNs {
```

Comment:
Doing this logic within `listClusterServiceARNs` and changing that to `listClusterServiceNames` seems to be more inline with other entity.Lists (e.g. environment's `listClusterNames`)

Reply:
From what I can tell in looking at `environment_list.go` and `load_balancer_list.go`, only the one line from this `for` loop would go into `listClusterServiceNames()`. Trimming off the Layer0 prefix and creating IDs from Names seems to consistently be done here in the master function.

Reply:
ah, good call